### PR TITLE
Histogram view added numeric label sort option

### PR DESCRIPTION
**Pull request by Atlas139mkm**
### Description of the Change

- Added a numeric sort option by label. Label (Number)
- Changed label option to Label (String)

Descending Number
image

Descending Number
image

Ascending Number
image

Ascending String
image

### Why Should This Be In Core?

Allows users to sort their labels by their numeric value if the labels can be converted to a number.

### Benefits

Potential use cases where the labels are numbers and the user would like to sort them in ascending or descending order.

### Verification Process

Screen shots provide various different mixes of numbers, strings, number/string combos. These show how it is working.

### Applicable Issues
#861

### CODE REVIEW

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/BinComparator.java [hunk 1]
```diff
@@ -200,6 +240,18 @@ public boolean usesSelection() {
         return usesSelection;
     }
 
+    public boolean isNumeric(String strNum) {
```
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/BinComparator.java:243]: parameter final

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/BinComparator.java [hunk 2]
```diff
@@ -200,6 +240,18 @@ public boolean usesSelection() {
...
+        if (strNum == null) {
+            return false;
+        }
+        try {
+            Float.parseFloat(strNum);
+        } catch (NumberFormatException nfe) {
```
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/BinComparator.java:249]: exception final

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/BinComparator.java [hunk 3]
```diff
@@ -25,10 +25,50 @@
...
+        }
+
+        @Override
+        public BinComparator getReverse() {
+            return REVERSE_KEY_NUMBER;
+        }
+    },
+    REVERSE_KEY_NUMBER("Label (Number ascending)", false, false) {
+        @Override
+        public int compare(Bin o1, Bin o2) {
```
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/BinComparator.java:50]: parameters final

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/BinComparator.java [hunk 4]
```diff
@@ -25,10 +25,50 @@
  */
 public enum BinComparator implements Comparator<Bin> {
 
+    KEY_NUMBER("Label (Number)", true, false) {
+        @Override
+        public int compare(Bin o1, Bin o2) {
```
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/BinComparator.java:30]: parameters final

**Review by antares1470 — CHANGES_REQUESTED** [review]
Seems to work well. If you could add a whatsnew, that would be great

**Review by antares1470 — APPROVED** [review]

**Review by arcturus2 — APPROVED** [review]
I'm going to approve this and merge it in but I will do a PR and include you in with a suggestion to change the label from "String" to "Text". String makes sense to devs but to a non technical person a "String" is a more as in "a piece of string".

**Pull request merged**